lucille-core/src/main/java/com/kmwllc/lucille/indexer/SolrIndexer.java

```diff
@@ -128,20 +108,26 @@ protected void sendToIndex(List<Document> documents) throws Exception {
         solrDocRequestsByCollection.put(collection, new SolrDocRequests());
       }
       SolrDocRequests solrDocRequests = solrDocRequestsByCollection.get(collection);
-      String solrId = collection != null ? collection : doc.getId();
+      String solrId = idOverride != null ? idOverride : doc.getId();
 
       if (isDeletion(doc)) {
 
         // if the add/update requests contain the ID of this delete, send the add/updates
         // immediately so the add/update
         // of the document is processed before this delete.
 
-        if (solrDocRequests.containsIdForAddUpdate(solrId)) {
+        if (solrDocRequests.containsIdForAddUpdate(solrId)
+            || (doc.has(deleteByFieldField) && doc.has(deleteByFieldValue))) {
           sendAddUpdateBatch(collection, solrDocRequests.getAddUpdateDocs());
           solrDocRequests.resetAddUpdates();
         }
 
-        solrDocRequests.addIdForDeletion(solrId);
+        if (doc.has(deleteByFieldField) && doc.has(deleteByFieldValue)) {
```

Comment:
Same comment as above, I think we need to check that deleteByFieldField and deleteByFieldValue are non-null before passing them to doc.has